btb/tuning/hyperparams/numerical.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+
+"""Package where the numerical hyperparameters are defined."""
+
+import sys
+
+import numpy as np
+
+from btb.tuning.hyperparams.base import BaseHyperParam
+
+
+class NumericalHyperParam(BaseHyperParam):
+    """Numerical Hyperparameter Class.
+
+    The numerical hyperparameter class defines an abstraction to hyperparameters which ranges are
+    defined by a numerical value and can take any number between that range.
+
+    Attributes:
+        K (int):
+            Number of dimensions that this hyperparameter uses to be represented in the search
+            space.
+
+    Args:
+        min (int or float):
+            Minimum numerical value that this hyperparameter can be set to.
+        max (int or float):
+            Maximum numerical value that this hyperparameter can be set to.
+        include_min (default=True):
+            Either ot include or not the ``min`` value inside the range.
+        include_max (default=True):
+            Either ot include or not the ``max`` value inside the range.
+        step (int or float):
+            Increase amount to take for each sample.
+    """
+    def _within_range(self, values, min=0, max=1):
+        if (values < min).any() or (values > max).any():
+            raise ValueError('Value not within range [{}, {}]: {}'.format(min, max, values))
+
+
+class FloatHyperParam(NumericalHyperParam):
+    """NumericalHyperParam of type ``float``.
+
+    Hyperparameter space:
+        ``[min, max]``
+
+    Search space:
+        ``[0, 1)``
+    """
+
+    K = 1
```

Comment:
This can be put in the parent class, as the value is the same for all the subclasses.